### PR TITLE
[Prim] Add p_norm_decomp

### PR DESCRIPTION
**Pull request by HydrogenSulfate**
### PR Category

Operator Mechanism

### PR Types

New features

### Description

Pcard-75624

Add `p_norm_decomp` as composite forwad API for `p_norm`

- [x] Add unitest

### CODE REVIEW

File: test/legacy_test/test_norm_all.py [hunk 1]
```diff
@@ -194,7 +207,7 @@ def test_check_output(self):
         self.check_output()
 
     def test_check_grad(self):
-        self.check_grad(['X'], 'Out')
+        self.check_grad(['X'], 'Out', check_prim=True)
```
Comment (cyber-pioneer) [inline test/legacy_test/test_norm_all.py]:
```suggestion
        self.check_grad(['X'], 'Out', check_prim_pir=True)
```

File: test/legacy_test/test_norm_all.py [hunk 2]
```diff
@@ -175,6 +186,8 @@ class TestPnormOp(OpTest):
     def setUp(self):
         self.op_type = "p_norm"
         self.python_api = p_norm_python_api
+        self.public_python_api = norm_public_python_api
+        self.prim_op_type = "prim"
```
Comment (cyber-pioneer) [inline test/legacy_test/test_norm_all.py]:
```suggestion
        self.prim_op_type = "comp"
```

File: paddle/fluid/primitive/composite/composite.h [hunk 3]
```diff
@@ -127,6 +127,40 @@ static bool valid_type(const DataType& dtype) {
...
+    res = elementwise_pow<T>(res, inv_porder_tensor);
+  }
+
+  if (need_cast) {
+    return cast<T>(res, org_dtype);
+  } else {
+    return res;
+  }
+}
+
```
Comment (cxxly) [inline paddle/fluid/primitive/composite/composite.h:183]: 建议组合算子签名和PHI/PIR签名保持一致，用namespace约束分类，函数名不要加_decomp。存量的可以后续统一修改

**Review by cyber-pioneer — APPROVED** [review]

**Review by Xreki — APPROVED** [review]
LGTM for the atol

**Pull request merged**